compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs

```diff
@@ -1955,6 +1964,410 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         false
     }
 
+    fn point_at_specific_expr_if_possible_for_obligation_cause_code(
+        &self,
+        obligation_cause_code: &traits::ObligationCauseCode<'tcx>,
+        expr: &'tcx hir::Expr<'tcx>,
+    ) -> Result<&'tcx hir::Expr<'tcx>, &'tcx hir::Expr<'tcx>> {
+        match obligation_cause_code {
+            traits::ObligationCauseCode::ImplDerivedObligation(impl_derived) => self
+                .point_at_specific_expr_if_possible_for_derived_predicate_obligation(
+                    impl_derived,
+                    expr,
+                ),
+            traits::ObligationCauseCode::ExprBindingObligation(_, _, _, _) => Ok(expr),
+            _ => {
+                // Err(expr) indicates that we cannot process any further.
+                Err(expr)
+            }
+        }
+    }
+
+    /// We want to achieve the error span in the following example:
+    ///
+    /// ```
```

Comment:
```suggestion
    /// ```ignore (just for demonstration)
```

... unless you want to fix the other unrelated errors in this snippet, and make it `compile_fail,E0277` or whatever, instead of `ignore`